.github/workflows/website.yml

```diff
@@ -48,30 +68,77 @@ jobs:
+            ;
+            echo 'NEXT_PUBLIC_MAGIC_TESTMODE_ENABLED="1"';
+          ) > .env
         env:
-          NEXT_PUBLIC_API: http://localhost:8787
-          NEXT_PUBLIC_ENV: test-e2e
           NEXT_PUBLIC_MAGIC: ${{ secrets.STAGING_MAGIC_PUBLIC_KEY }}
+          MAGIC_SECRET_KEY: ${{ secrets.STAGING_MAGIC_SECRET_KEY }}
+      - name: load db schema
+        run: |
+          npm -w '@web3-storage/db' run start
+          npm -w '@web3-storage/db' run load-schema
+          npm -w '@web3-storage/db' run stop
+      - run: node -r dotenv/config $(which npm) run build -w packages/website
       - run: npx playwright install --with-deps ${{ matrix.browser }} chromium
         working-directory: packages/website
-      - run: touch .env
-      - run: xvfb-run --auto-servernum --server-args="-screen 0 1280x960x24" -- npm run -w @web3-storage/website test:e2e -- --project=${{ matrix.browser }}
+      - id: website_test_e2e
+        run: xvfb-run --auto-servernum --server-args="-screen 0 1280x960x24" -- node -r dotenv/config $(which npm) run -w @web3-storage/website test:e2e -- --project=${{ matrix.browser }} --reporter=html
         env:
-          NEXT_PUBLIC_API: http://localhost:8787
-          NEXT_PUBLIC_ENV: dev
-          NEXT_PUBLIC_MAGIC: ${{ secrets.STAGING_MAGIC_PUBLIC_KEY }}
-      - name: Check for website test results
+          PLAYWRIGHT_TIMEOUT: 120000
+
+      - if: always()
+        name: Check for website playwright-report
+        id: check_website_playwright_report
+        uses: andstor/file-existence-action@v1
+        with:
+          files: packages/website/playwright-report
+
+      - name: add playwright-report to web3.storage
+        if: inputs.add_playwright_report_to_web3_storage && steps.check_website_playwright_report.outputs.files_exists == 'true'
+        uses: web3-storage/add-to-web3@v2
+        id: ipfs
+        with:
+          path_to_add: packages/website/playwright-report
+          web3_token: ${{ secrets.WEB3_TOKEN }}
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: log playwright-report on web3.storage cid, url
+        if: inputs.add_playwright_report_to_web3_storage && steps.check_website_playwright_report.outputs.files_exists == 'true'
+        run: |
+          echo cid: ${{ steps.ipfs.outputs.cid }}
+          echo url: https://nftstorage.link/ipfs/${{ steps.ipfs.outputs.cid }}
+      - id: create_output_playwright_reporter_url
+        if: inputs.add_playwright_report_to_web3_storage && steps.check_website_playwright_report.outputs.files_exists == 'true'
+        run: echo "::set-output name=url::https://nftstorage.link/ipfs/${{ steps.ipfs.outputs.cid }}"
+      - id: create_output_playwright_reporter_cid
+        if: inputs.add_playwright_report_to_web3_storage && steps.check_website_playwright_report.outputs.files_exists == 'true'
+        run: echo "::set-output name=cid::${{ steps.ipfs.outputs.cid }}"
+
+      - name: upload playwright-report artifact
```

Comment:
I think you could drop a whole bunch of `if` logic from the steps above by having the e2e test run job only do the uploading of the playwright-report (where it exists)... you can a dependent job to publish the report and have that fetch the artefact from GitHub, and skip the whole job if there is no report to publish

Reply:
it takes so long to get to this part of the build so its time-expensive to iterate on refactors like that :/

Reply:
lookin forward to future of tooling like this https://dagger.io/

Reply:
I cleaned up the uses of `if`

I also evaluated deferring the playwright-report storage to another job. Because test-e2e uses a matrix to test across more than one browser, so would any other job, which I think would make the code a bit more complex (keeping 'matrix' in sync across several jobs).